MapleServer2/Packets/ShopPacket.cs

```diff
@@ -0,0 +1,161 @@
+            {
+                pWriter.WriteInt(product.UniqueId); // 968620
+                pWriter.WriteInt(product.ItemId);
+                pWriter.WriteByte(product.TokenType); // Currency Type
+                pWriter.WriteInt(product.RequiredItemId); // Only used when tokenType is type Capsule (1)
+                pWriter.WriteInt();
+                pWriter.WriteInt(product.Price); // Current Price (e.g discounted price)
+                pWriter.WriteInt(product.SalePrice); // Original Price
+                pWriter.WriteByte(product.ItemRank); // Stars
+                pWriter.WriteUInt(0xEFDA5D2D);
+                pWriter.WriteInt(product.StockCount); // Total Stock
+                pWriter.WriteInt(product.StockPurchased); // Purchased Stock (remaining stock will show Total - Purchased)
+                pWriter.WriteInt(product.GuildTrophy); // "More than x guild trophies"
+                pWriter.WriteMapleString(product.Category); // Item Category (found in the item's xml)
+                pWriter.WriteInt(product.RequiredAchievementId); 
+                pWriter.WriteInt(product.RequiredAchievementGrade); // When > 0: "You must have the "X. TrophyNameHere" trophy to do this."
+                pWriter.WriteByte(product.RequiredChampionshipGrade); // Guild ranking above X
+                pWriter.WriteShort(product.RequiredChampionshipJoinCount); // Must participate more than X times
+                pWriter.WriteByte(product.RequiredGuildMerchantType); // 2 = "Guild Supply Merchant" 3 = "Guild Gemstone Merchant"
+                pWriter.WriteShort(product.RequiredGuildMerchantLevel); // The guild <type> merchant must be above level X
+                pWriter.WriteBool(false);
+                pWriter.WriteShort(product.Quantity); // Bundle Quantity
+                pWriter.WriteByte(1);
+                pWriter.WriteByte(product.Flag); // New, Sale, Event, Hot, etc.
+                pWriter.WriteByte();
+                pWriter.WriteShort(product.RequiredQuestAlliance); // Required faction
+                pWriter.WriteInt(product.RequiredFameGrade); // Required reputation for the above faction type
+                pWriter.WriteBool(false);
+                
+                // Write item data
+                pWriter.WriteByte();
+                pWriter.WriteByte();
+                pWriter.WriteInt(1); // Unknown (amount?)
+                pWriter.WriteInt();
+                pWriter.WriteInt(-1);
+                pWriter.WriteLong(DateTimeOffset.UtcNow.ToUnixTimeSeconds()); // Item creation time
+                pWriter.WriteZero(52);
+                pWriter.WriteInt(-1);
+                pWriter.WriteZero(102);
+                pWriter.WriteInt(1);
+                pWriter.WriteZero(28);
+                pWriter.WriteLong(1); // Item owner character id (shop items have no owner)
+                pWriter.WriteShort(); // Item owner name (shop items have no owner)
+                pWriter.WriteZero(12);
```

Comment:
Can't this be `pWriter.WriteItem(item)`? It looks like the same logic.

Reply:
Same as above. Unfortunately it's slightly different :(